Base/WebService.swift

```diff
@@ -12,38 +12,147 @@ public protocol UnauthorizedResponseHandler {
     func authorizedRequestDidFail(request: URLRequest, response: HTTPURLResponse, data: Data?, retry: @escaping () -> Void)
 }
 
+struct DataTaskCallback<A> {
+    let success: (URLResponse?, A) -> Void
+    let failure: (Error) -> Void
+    let retry: () -> Void
+    var unauthorizedResponseHandler: UnauthorizedResponseHandler?
+}
+
+extension URLSession {
+    
+    private func isCancelledError(_ error: NSError) -> Bool {
```

Comment:
Strange for this and the next function to be instance methods on a session, should be an extension on Error or at least a static func somewhere